### PR TITLE
Adding tag support

### PR DESCRIPTION
**Pull request by AlienHoboken**
Added tag support for DO Api, including listing and deleting droplets by tag.

### CODE REVIEW

File: DigitalOcean.API/Clients/TagsClient.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            };
+            return _connection.ExecuteRequest<Tag>("tags/{name}", parameters, null, "tag");
+        }
+
+        /// <summary>
+        /// Create a new Tag
+        /// </summary>
+        public Task<Tag> Create(string tagName) {
+            Models.Requests.Tag data = new Models.Requests.Tag();
+            data.Name = tagName;
```
Comment (vevix) [inline DigitalOcean.API/Clients/TagsClient.cs]: Swap this for an Object Initializer

```csharp
var data = new Models.Requests.Tag {
    Name = tagName
};
```

File: DigitalOcean.API/Clients/TagsClient.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+            return _connection.ExecuteRequest<Tag>("tags", null, data, "tag", Method.POST);
+        }
+
+        /// <summary>
+        /// Update the name of a Tag
+        /// </summary>
+        public Task<Tag> Update(string tagName, string newTagName) {
+            Models.Requests.Tag data = new Models.Requests.Tag();
+            data.Name = newTagName;
```
Comment (vevix) [inline DigitalOcean.API/Clients/TagsClient.cs]: Swap this for an Object Initializer

File: DigitalOcean.API/Clients/TagsClient.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+            return _connection.ExecuteRequest<Tag>("tags/{name}", parameters, data, "tag", Method.PUT);
+        }
+
+        /// <summary>
+        /// Tag existing resources of given resource id / type combination
+        /// </summary>
+        public Task Tag(string tagName, List<KeyValuePair<string, string>> resources) {
+            Models.Requests.TagResource data = new Models.Requests.TagResource();
+            data.Resources = new List<Models.Requests.TagResource.Resource>();
```
Comment (vevix) [inline DigitalOcean.API/Clients/TagsClient.cs]: Swap this for an Object Initializer

File: DigitalOcean.API/Clients/TagsClient.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+            return _connection.ExecuteRaw("tags/{name}/resources", parameters, data, Method.POST);
+        }
+
+        /// <summary>
+        /// Untag existing resources of given resource id / type combination
+        /// </summary>
+        public Task Untag(string tagName, List<KeyValuePair<string, string>> resources) {
+            Models.Requests.TagResource data = new Models.Requests.TagResource();
+            data.Resources = new List<Models.Requests.TagResource.Resource>();
```
Comment (vevix) [inline DigitalOcean.API/Clients/TagsClient.cs]: Swap this for an Object Initializer

**Comment by vevix** [comment]
Great work! :100:

Could you go thru and fix the style issues so the codebase is consistent? I left some comments, but there is also a few mismatching opening brackets (Codebase follows [1TBS](https://en.wikipedia.org/wiki/Indent_style#Variant:_1TBS_.28OTBS.29))

**Comment by AlienHoboken** [comment]
Missed those object initializers and a few brackets, thanks for pointing it out! Fixed the style issues, even found a couple that weren't mine and corrected. 😊

**Comment by vevix** [comment]
Thanks! I'll merge this when I'm ready to push a new version to NuGet. Hopefully tonight.

**Pull request merged**